public/electron.ts

```diff
@@ -37,14 +47,31 @@ export async function createWindow(): Promise<void> {
   mainWindow.on('closed', (): void => {
     mainWindow = undefined;
   });
-  mainWindow.webContents.on('will-navigate', (event: Event, url: string): void => {
+  mainWindow.webContents.on('new-window', (event: Event, url: string): void => {
     if (url.startsWith('http:') || url.startsWith('https:')) {
       event.preventDefault();
       shell.openExternal(url);
     }
   });
 }
 
+function handleFileDialogs(): void {
+  ipcMain.on('did-start-file-save-dialog', async (event: IpcMainEvent, options: SaveDialogOptions): Promise<void> => {
+    if (mainWindow != null) {
+      winston.log('info', 'File save dialog');
+      const filePath = (await dialog.showSaveDialog(mainWindow, options)).filePath || '';
```

Comment:
Since I was mostly moving existing code I didn't think to much about the current implementation but a good question. The type of `filepath` is `string | undefined` so this makes sure its is defined. It may be more useful to allow the undefined state and change the type everywhere its used on the frontend and require it to deal with it but that is a refactor without a current need.

By contrast `filePaths` (plural) below has type `string[]` and is always defined.

Reply:
Fair enough. I didn't realise this was existing code until later in the PR. Ignore for now.